src/evict/README.md

```diff
@@ -0,0 +1,49 @@
+# Eviction
```

Comment:
Possibly a stupid question, but is `README.md` the right place for this? How does our doc generator (doxygen?) generate source.wiredtiger.com pages? Should we figure out how to have it create content there?

Reply:
Also fine for that to be a future problem

Reply:
We've decided to carve out a distinction between dev facing docs (this README.md) and user facing docs in doxygen. 
The line is fuzzy and we'll see overlap, but doxygen should entail "what does configuring eviction mean for my Mongo workload" and the README.md talks about specific details like pausing eviction to access some shared resource